From 65ce353b02e6259986f3541077fcc053885085ee Mon Sep 17 00:00:00 2001
From: Fugang Duan <b38611@freescale.com>
Date: Wed, 20 Jan 2016 16:33:54 +0800
Subject: [PATCH 1454/1691] MLK-12294 tty: serial: imx: init sport->flags
 before tx DMA ready

commit 65ce353b02e6259986f3541077fcc053885085ee from
git://git.freescale.com/imx/linux-2.6-imx.git imx_4.1.15_1.0.0_ga

Before then DMA tx path, init the DMA tx path synchronal flags
sport->flags.

Signed-off-by: Fugang Duan <B38611@freescale.com>
---
 drivers/tty/serial/imx.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/tty/serial/imx.c b/drivers/tty/serial/imx.c
index f9949b5..c9bd603 100644
--- a/drivers/tty/serial/imx.c
+++ b/drivers/tty/serial/imx.c
@@ -1065,6 +1065,7 @@ static void imx_enable_dma(struct imx_port *sport)
 	unsigned long temp;
 
 	init_waitqueue_head(&sport->dma_wait);
+	sport->flags = 0;
 
 	/* set UCR1 */
 	temp = readl(sport->port.membase + UCR1);
-- 
1.9.1

